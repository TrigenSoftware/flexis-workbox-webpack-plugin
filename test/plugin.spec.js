import path from 'path';
import compile, {
	fs,
	pathToArtifacts
} from './compile';

describe('workbox-webpack-plugin', () => {

	it('should emit precache manifest and modificate service worker', async () => {

		const stats = await compile('serviceWorkerLoader.js', {});
		const precacheManifestName = stats.assets
			.find(({ name }) => name.startsWith('precache-manifest'))
			.name;
		const precacheManifest = fs.readFileSync(
			path.join(pathToArtifacts, precacheManifestName),
			'utf8'
		);
		const serviceWorker = fs.readFileSync(
			path.join(pathToArtifacts, 'serviceWorker.js'),
			'utf8'
		);

		expect(precacheManifest).toMatchSnapshot();
		expect(serviceWorker).toEqual(
			expect.stringMatching(/^importScripts\("\/precache-manifest\./)
		);
	});
});
