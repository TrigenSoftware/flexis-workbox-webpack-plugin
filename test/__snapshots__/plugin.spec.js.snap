// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workbox-webpack-plugin should emit precache manifest and modificate service worker 1`] = `
"self.__precacheManifest = [
  {
    \\"revision\\": \\"5daa290cdfc55a04da4c1a6f37c6ba9a\\",
    \\"url\\": \\"/index.html\\"
  },
  {
    \\"url\\": \\"/bundle.e740afa204b94fafbe35.js\\"
  }
];"
`;
